### PR TITLE
Code decoder PR

### PR DESCRIPTION
**Pull request by mertkantaroglu**
Hello,

In this project me and @Nezerwa;
- [ ] Created required files for ruby and linter check
- [ ] Created a method to decode a Morse code character
- [ ] Created a method to decode an entire word in Morse code
- [ ] Created a method to decode an entire sentence in Morse code
- [ ] Followed Gitflow
- [ ] Fixed linter error using Rubocop

Looking forward to your response.

Thanks.

### CODE REVIEW

File: decode_code.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  decoded_chars.join(' ')
+end
+
+puts decode_word('-- -.--   -. .- -- .')
+
+# Decode a Sentence Using decode_word function
+def decode_sentence(sentence)
+  decoded_words = sentence.split.map { |word| decode_word(word) }
+  decoded_words.join(' ')
+end
```
Comment (franclobo) [inline decode_code.rb]: - [x] I recommend changing the code to distinguish the words between the spaces, and pay attention to the spaces on the join method, in word there aren't spaces, but there are in sentences. 👇🏽 

```ruby
def decode_word(word)
  decoded_chars = word.split.map { |char| decode_char(char) }
  decoded_chars.join('')
end

puts decode_word('-- -.--   -. .- -- .')

# Decode a Sentence Using decode_word function
def decode_sentence(sentence)
  decoded_words = sentence.split('   ').map { |word| decode_word(word) }
  decoded_words.join(' ')
end
puts decode_sentence('.-  -... --- -..-   ..-. ..- .-.. .-..   --- ..-.  .-. ..- -... .. . ...')
```

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @mertkantaroglu and @Nezerwa, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct git flow ✔️ 
- Good decode methods ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decode_code.rb (hunk 1 above)
Comment (mertkantaroglu) [reply decode_code.rb]: Hi @franclobo . 

I am not sure I understand what you mean. Can you please clarify your comment and give some example?

Thanks.

File: decode_code.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# Decode a Word Using decode_char function
+def decode_word(word)
+  word.split.map { |char| decode_char(char) }.join
+end
+
+puts decode_word('-- -.--   -. .- -- .')
+
+# Decode a Sentence Using decode_word function
+def decode_sentence(sentence)
+  sentence.split('  ').map { |word| get_word(word) }.join('  ')
```
Comment (Davidosky007) [inline decode_code.rb]: * Kindly fix this error the method `get_word` is not defined I think the right method to use there will be `decode_word` 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @mertkantaroglu @Nezerwa 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Ruby syntax used properly 👍 
* No linter errors 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decode_code.rb (hunk 2 above)
Comment (mertkantaroglu) [reply decode_code.rb]: Thanks @Davidosky007 I just fixed it.

**Review by Davidosky007 — APPROVED** [review]
Hi @mertkantaroglu @Nezerwa 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Proper use of the ruby syntax 👍 
* Good looking readme 👍 
* No linter errors 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mertkantaroglu** [comment]
Thank you @Davidosky007 👍

**Pull request merged**